types/lookup_builder.go

```diff
@@ -8,26 +8,81 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"net/url"
 
 	"github.com/openfaas-incubator/connector-sdk/types"
-	"github.com/openfaas/faas/gateway/requests"
+	ptypes "github.com/openfaas/faas-provider/types"
+	"github.com/pkg/errors"
 )
 
 // FunctionLookupBuilder alias for types.FunctionLookupBuilder
 type FunctionLookupBuilder types.FunctionLookupBuilder
 
-// GetFunctions requests the OpenFaaS gteway to return a list of all functions
-func (lookup *FunctionLookupBuilder) GetFunctions() ([]requests.Function, error) {
-	req, _ := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/system/functions", lookup.GatewayURL), nil)
+//GetNamespaces get openfaas namespaces
+func (s *FunctionLookupBuilder) GetNamespaces() ([]string, error) {
```

Comment:
Please use CLI SDK for this API calls, try to vendor `github.com/openfaas/faas-cli/proxy` and use API to get namespaces.

Reply:
Thanks! I've removed it, and now using the CLI SDK.